core/src/main/java/io/grpc/inprocess/InProcessTransport.java

```diff
@@ -806,6 +807,10 @@ public void setDeadline(Deadline deadline) {
         long effectiveTimeout = max(0, deadline.timeRemaining(TimeUnit.NANOSECONDS));
         headers.put(TIMEOUT_KEY, effectiveTimeout);
       }
+
+      @Override
+      public void appendTimeoutDetails(ToStringHelper toStringHelper) {
```

Comment:
TBH, I think StringBuilder is a better fit.  

```java
public void appendTimeoutDetails(StringBuilder buffer) {
  buffer.append(key).append('=').append(value).append(',')
}
```

The code that finally does toString can trim the last character.  Alternatively, this could just be a `Map<String, Object>`.